src/server.c

```diff
@@ -3720,6 +3729,15 @@ static void handle_join(int cid, char *ptr)
 		send_msgf(cid, MSG_JOINNAK, "s", "Incorrect game password");
 		return;
 	}
+	/* Check for a version compatible with Xeno Invasion expansion */
```

Comment:
An old client can connect to a new server, and such client shouldn't join XI channels.

The old clients won't see the XI channels, so clients running our implementation won't try to join such channels. It is also true that I first added the test with the intention of protecting the old client state. Indeed, such clients do no need this protection anymore.

OK i'll remove it.
done